pkg/kubecost/allocationprops.go

```diff
@@ -308,7 +308,7 @@ func (p *AllocationProperties) GenerateKey(aggregateBy []string, labelConfig *La
 			if annotations == nil {
 				names = append(names, UnallocatedSuffix)
 			} else {
-				annotationName := labelConfig.Sanitize(strings.TrimPrefix(agg, "annotation:"))
```

Comment:
example (same for lables):
agg = "a/b"
(before) annotationName = "a_b"
annotations map contains "a/b", not "a_b", therefore allocation is aggregated to __unallocated__